src/battle_setup.c

```diff
@@ -1846,7 +1862,11 @@ static bool32 HasAtLeastFiveBadges(void)
 
 void IncrementRematchStepCounter(void)
 {
-    if (HasAtLeastFiveBadges())
+    if (HasAtLeastFiveBadges()
+#if I_VS_SEEKER_CHARGING != 0
+        && (!CheckBagHasItem(ITEM_VS_SEEKER, 1))
+#endif
+       )
```

Comment:
```suggestion
    if (HasAtLeastFiveBadges() && (I_VS_SEEKER_CHARGING == 0 || (!CheckBagHasItem(ITEM_VS_SEEKER, 1))))
```

Reply:
Fixed in https://github.com/PokemonSanFran/pokeemerald-expansion/commit/e4fe0e19eff65df5e93af055551d598e7ff4fcb8